streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -153,10 +151,17 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
         final KTableSource<K, V> tableSource = new KTableSource<>(storeSupplier.name());
 
 
-        final Deserializer<K> keyDeserializer = keySerde == null ? null : keySerde.deserializer();
-        final Deserializer<V> valueDeserializer = valSerde == null ? null : valSerde.deserializer();
+        final Deserializer<K> keyDeserializer = consumed.keySerde() == null ? null : consumed.keySerde().deserializer();
```

Comment:
Just for clarifying: we are giving the serde twice here, onces in the store supplier and once in `addGlobalStore`. I think the latter will be cleared by the end of KIP-182?

Reply:
@guozhangwang the supplier will eventually be replaced with `Materialized` - it will have serdes though